### PR TITLE
fix: rebroadcast txs recovered from reorg cache

### LINKED ISSUE #3489: Increase "reorg cache" timer limit?
We should increase for every node the reorg_cache timer limit which is set by default to 30 minutes if it affects the mempool, so in case of future large reorg, nodes will rebroadcast automatically all unconfirmed transactions which were confirmed in the stale blocks.

If it only affects our local txs pool, we should consider to add the mempool in the reorg cache for the reason mentioned above.

### PR DESCRIPTION
**Pull request by iho**
## Summary

On reorg, the reorg cache correctly re-adds previously seen txs to the **local** mempool, but it never rebroadcast them. Peers that did not keep those txs (or never saw them) would not learn about them until someone else re-announced.

As noted on #3489 by @antiochp: every tx successfully re-accepted from the reorg cache should also be treated as new and rebroadcast.

Closes #3489

### Changes
- `reconcile_reorg_cache`: call `adapter.tx_accepted` after a successful `add_to_txpool`
- Skip rebroadcast when re-add fails (already in pool / invalid on new tip)
- Integration test with a recording pool adapter

Retention period remains configurable via `reorg_cache_period` (default 30 minutes). This PR focuses on the missing rebroadcast path rather than changing the default window.

## Test plan
- [x] `cargo test -p grin_pool --test reorg_cache_rebroadcast`

### CODE REVIEW

File: pool/src/transaction_pool.rs [hunk 1]
```diff
@@ -283,19 +283,31 @@ where
 		debug!("truncate_reorg_cache: size: {}", cache.len());
 	}
 
+	/// Re-apply cached transactions to the mempool after a reorg.
+	///
+	/// Txs that are successfully re-accepted are also rebroadcast via the pool
```
Comment (wiesche89) [inline pool/src/transaction_pool.rs:288]: The later discussion in #3489 explicitly questions unconditional rebroadcasting and suggests opt-in/default-off. This PR also leaves the issue’s retention-period request unchanged. Could we settle that policy here and use Refs #3489 rather than closing it unless both parts are handled?

File: pool/src/transaction_pool.rs [hunk 2]
```diff
@@ -283,19 +283,31 @@ where
...
 			entries.len(),
 			header.hash(),
 		);
+		let mut rebroadcast = 0usize;
 		for entry in entries {
-			let _ = self.add_to_txpool(&entry, header);
+			// Only rebroadcast txs that were actually re-accepted (already-present
+			// or invalid-on-new-tip txs are skipped without network spam).
+			if self.add_to_txpool(&entry, header).is_ok() {
+				self.adapter.tx_accepted(&entry);
```
Comment (wiesche89) [inline pool/src/transaction_pool.rs:303]: tx_accepted() is fine for a single tx, but here it runs in a loop under the pool write lock. Each peer queue has at most 100 free slots and silently drops overflow while reporting success, so a reorg recovering more than SEND_CHANNEL_CAP txs can lose later announcements. Could we schedule these through a paced bulk-relay path outside the pool lock?

File: pool/tests/reorg_cache_rebroadcast.rs [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+		.unwrap();
+	pool.add_to_pool(test_source(), tx_b.clone(), false, &header)
+		.unwrap();
+	// Initial accept calls: one per tx.
+	assert_eq!(adapter.accepted_count(), 2);
+	assert_eq!(pool.total_size(), 2);
+	assert_eq!(pool.reorg_cache.read().len(), 2);
+
+	// Simulate reorg: empty the pool while leaving reorg_cache intact.
+	pool.txpool.entries.clear();
```
Comment (wiesche89) [inline pool/tests/reorg_cache_rebroadcast.rs:72]: The recording adapter only counts callbacks. Could we add a real fork/reorg test for the production trigger, plus a bounded-queue adapter using SEND_CHANNEL_CAP to cover overflow? Otherwise the test misses the actual delivery failure mode.

File: pool/tests/reorg_cache_rebroadcast.rs [hunk 4]
```diff
@@ -0,0 +1,97 @@
+// Copyright 2021 The Grin Developers
```
Comment (wiesche89) [inline pool/tests/reorg_cache_rebroadcast.rs:1]: 2026?

File: pool/tests/common.rs [hunk 5]
```diff
@@ -170,6 +170,57 @@ where
...
+impl PoolAdapter for RecordingPoolAdapter {
+	fn tx_accepted(&self, entry: &PoolEntry) {
+		self.accepted.lock().unwrap().push(entry.tx.hash());
+	}
+	fn stem_tx_accepted(&self, _entry: &PoolEntry) -> Result<(), PoolError> {
+		Ok(())
+	}
+}
+
+pub fn init_transaction_pool_with_adapter<B, P>(
```
Comment (wiesche89) [inline pool/tests/common.rs:203]: this duplicates the full PoolConfig literal from init_transaction_pool for one caller. Could the existing helper delegate here, or both use a shared test config, so they cannot drift?
